bitrise.yml

```diff
@@ -1,20 +1,68 @@
----
-format_version: '8'
+format_version: 11
 default_step_lib_source: https://github.com/bitrise-io/bitrise-steplib.git
 project_type: other
 
 # To customize how workflows are run on different triggers,
 # go to Workflow Editor on Bitrise.io.
 
 workflows:
-  push_to_any_branch:
+  push_to_pull_request:
+    description: |-
+        This workflow is triggered on starting new PR or pushing new changes to existing PRs.
+        By default, it doesn't run any tests, but this behaviour is overwritten in `choose_workflows.py` when:
+        - one or more `DD_OVERWRITE_RUN_*` ENVs are passed to the current CI job:
+            - DD_OVERRIDE_RUN_UNIT_TESTS='1' to force running unit tests
+            - DD_OVERRIDE_RUN_INTEGRATION_TESTS='1' to force running integration tests
+            - DD_OVERRIDE_RUN_SMOKE_TESTS='1' to force running smoke tests
+            - DD_OVERRIDE_TOOLS_TESTS='1' to force running tools tests
+        - a type of tests is selected in the PR description checklist,
+        - PR modifies files matching given filter (e.g. changing a file in `Sources/*` will trigger unit tests)
+    envs:
+      - DD_RUN_UNIT_TESTS: '0'
+      - DD_RUN_INTEGRATION_TESTS: '0'
+      - DD_RUN_SMOKE_TESTS: '0'
+      - DD_RUN_TOOLS_TESTS: '0'
+    after_run:
+    - _make_dependencies
+    - run_conditioned_workflows
+    - _deploy_artifacts
+
+  push_to_develop_or_master:
+    description: |-
+        This workflow is triggered for each new commit pushed to `develop` or `master` branch.
+        It runs unit and integration tests and reports failures to Slack.
+    envs:
+      - DD_RUN_UNIT_TESTS: '1'
+      - DD_RUN_INTEGRATION_TESTS: '1'
+      - DD_RUN_SMOKE_TESTS: '0'
+      - DD_RUN_TOOLS_TESTS: '0'
+    after_run:
+    - _make_dependencies
+    - run_conditioned_workflows
+    - _deploy_artifacts
+    - _notify_failure_on_slack
+
+  run_nightly_smoke_and_tools_tests:
+    description: |-
+        This workflow is triggered every night.
+        It runs smoke and tools tests and reports failures to Slack.
+    envs:
+      - DD_RUN_UNIT_TESTS: '0'
+      - DD_RUN_INTEGRATION_TESTS: '0'
+      - DD_RUN_SMOKE_TESTS: '1'
+      - DD_RUN_TOOLS_TESTS: '1'
     after_run:
     - _make_dependencies
-    - run_linter
-    - run_unit_tests
-    - run_integration_tests
-    - check_dependency_managers
+    - run_conditioned_workflows
     - _deploy_artifacts
+    - _notify_failure_on_slack
+
+
+  # Temporary workflow for enabling backward compatibility for pre- RUMM-2110 triggers.
+  # TODO: remove this workflow and clean-up triggers when RUMM-2110 is merged into `master` with `1.11.0`.
+  push_to_any_branch:
+    after_run:
+    - push_to_pull_request
```

Comment:
I still need to figure out how to redirect bitrise.io triggers into these new workflows. Ideally, both new and old triggers need to work together (for the sake of doing work on `master` or raising a hotfix).